### ISSUE TITLE
Fix public-api rule: Remove file extensions when suggesting directory imports

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#17]
## Bug Description

The `public-api` rule incorrectly handles file extensions in fix suggestions when converting from specific file imports to directory imports.

## Current Behavior

When importing from:
```typescript
import { unblockNode, unblockNodesBulk } from "@/entities/node/api.ts";
```

The rule suggests:
```typescript
import { unblockNode, unblockNodesBulk } from "@/entities/node.ts";
```

## Expected Behavior

The rule should suggest:
```typescript
import { unblockNode, unblockNodesBulk } from "@/entities/node";
```

## Root Cause

In `src/rules/public-api/model/convert-to-public-api.ts`, the `convertToPublicApi` function removes the segment path (e.g., `/api.ts`) but doesn't account for the case where the remaining path still has a file extension that should be removed when the result is a directory reference.

## Reproduction

```typescript
// Input
import { something } from "@/entities/node/api.ts";

// Current incorrect output  
import { something } from "@/entities/node.ts";

// Expected correct output
import { something } from "@/entities/node";
```

## Proposed Solution

The `convertToPublicApi` function should check if the resulting path ends with a file extension (`.ts`, `.js`, etc.) and remove it when the result is meant to be a directory import.

## Files Affected

- `src/rules/public-api/model/convert-to-public-api.ts` - Main logic
- `src/rules/public-api/index.test.ts` - Add test case

**Comment by conarti** [comment]
Fixed in 2.0.0-rc.16 (2a1e9c3 f4a91c3). You can try it with `npm i -D @conarti/eslint-plugin-feature-sliced@next`; it will ship in 2.0.0.
